### PR TITLE
[CALCITE-6178] WITH RECURSIVE query when cloned using sqlshuttle loos…

### PR DESCRIPTION
**Pull request by HanumathRao**
…es RECURSIVE property

Fix the getters and setters of SqlWithItem to pass recursive flag as one of the operand.

### CODE REVIEW

File: testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java [hunk 1]
```diff
@@ -1233,6 +1235,17 @@ protected static SortedSet<String> keywords(@Nullable String dialect) {
         .fails("Bang equal '!=' is not allowed under the current SQL conformance level");
   }
 
+  @Test void testRecursiveQueryCloned() throws Exception {
```
Comment (mihaibudiu) [inline testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java:1238]: May be useful to add a Javadoc comment here with the issue that is being solved.

**Review by mihaibudiu — APPROVED** [review]

File: testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java (hunk 1 above)
Comment (HanumathRao) [reply testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java:1241]: Thanks @mihaibudiu for the review, added a Javadoc comment to the test case.

**Comment by mihaibudiu** [comment]
If no one else has comments I think we should merge this PR

**Comment by mihaibudiu** [comment]
@HanumathRao if you can squash the commits we can merge this PR. Thank you.

**Comment by HanumathRao** [comment]
> @HanumathRao if you can squash the commits we can merge this PR. Thank you.

Done. Thank you.

**Pull request merged**